docs/vectorizer_guide.md

```diff
@@ -0,0 +1,243 @@
+```
+
+You can combine this with metadata filters by adding a WHERE clause:
+
+```sql
+SELECT 
+   chunk,
+   embedding <=> <query embedding> as distance
+FROM blog_contents_embeddings
+WHERE 
+   metadata->>'department' = 'finance'
+ORDER BY 
+   distance 
+LIMIT 10;
+```
+
+This approach works with any column from the blog table. For example, to search by author:
+
+```sql
+SELECT 
+   chunk,
+   embedding <=> <query embedding> as distance,
+   author
+FROM blog_contents_embeddings       
+WHERE 
+   author = 'Bulgakov'    
+ORDER BY 
+   distance 
+LIMIT 10;
+```
+
+## Advanced vectorizer usage
+
+### Formatting 
+
+Formatting allows you to inject additional information into each chunk. This is
+needed because splitting up the text into chunks can lead to a loss of important
+context. For instance, you might want to include the authors and title with each
+chunk. This is achieved using Python template strings, which have access to all
+columns in the row and a special '$chunk' variable containing the chunk's text.
+
+You may need to reduce the chunk size to ensure the formatted text fits within
+token limits. Adjust the 'chunk_size' parameter of the text_splitter
+accordingly:
+
+```sql
+SELECT ai.create_vectorizer(
+    'blog'::regclass,
+    destination => 'blog_contents_embeddings',
+    embedding => ai.embedding_openai('text-embedding-3-small', 768),
+    chunking => ai.chunking_character_text_splitter('contents', chunk_size => 700),
+    formatting => ai.formatting_python_template('$title - by $author - $chunk'),
+);
+```
+
+The default format string is simply '$chunk'.
+
+### Indexing
+
+The vectorizer can create a vector index on the embedding column to improve
+query performance. By default, a vectorscale index is created after 100,000
+rows, but you can specify other vector index types. Here's an example using an
+HNSW index:
+
+```sql
+SELECT ai.create_vectorizer(
+    'blog'::regclass,
+    destination => 'blog_contents_embeddings',
+    embedding => ai.embedding_openai('text-embedding-3-small', 768),
+    chunking => ai.chunking_character_text_splitter('contents', chunk_size => 700),
+    formatting => ai.formatting_python_template('$title - by $author - $chunk'),
+    indexing => ai.indexing_hnsw(min_rows => 100000, opclass => 'vector_l2_ops')
+);
+```
+
+Note: Indexing relies on a background job that runs periodically, so this
+feature won't work if scheduling is disabled.
+
+### Scheduling
+
+By default, scheduling uses TimescaleDB background jobs running every five
+minutes. You can disable this to run manually or through an external cron job:
+
+```sql
+SELECT ai.create_vectorizer(
+    'blog'::regclass,
+    destination => 'blog_contents_embeddings',
+    embedding => ai.embedding_openai('text-embedding-3-small', 768),
+    chunking => ai.chunking_character_text_splitter('contents', chunk_size => 700),
+    formatting => ai.formatting_python_template('$title - by $author - $chunk'),
+    scheduling => ai.scheduling_none(),
```

Comment:
`ai.scheduling_none()` requires `indexing=>ai.indexing_none()` otherwise an error is raised